src/dotnet/ToolPackage/CommandNameAllReservedValidator.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.DotNet.ToolPackage
+{
+    public static class CommandNameAllReservedValidator
+    {
+        private static CommandNameValidator[] _explicitReservedStrings = new CommandNameValidator[] {
```

Comment:
whole unit test for validation take about 200ms. And it is one off during dotnet install (which usually has networking call to download package). I don't think this affect perf. To have code matches the spec make it easier to edit

Reply:
200ms sounds like a lot to me. What's the percentage of dotnet tool install?

Reply:
the whole test is 200ms. with it around 200ms, without it around 200ms. So the actual increase is minimal.

Reply:
Oh, sorry I misread the original comment. 7ms is much more reasonable.

I withdraw the perf concern, but I'm still worried about the complexity here and how it will be easy to trigger false positives. That's feedback for the proposal not the implementation here, though.